File: 04_OJ/oj_leetcode_70/LC933.cpp
/**
 * @file LC933.cpp
 * @author liuyang
 * @brief 最近的请求次数 The Number of Calls
 * 
 * 题目：https://leetcode-cn.com/problems/number-of-recent-calls/
 * OJ状态：通过
 * 
 * @version 0.1
 * @date 2021-12-02
 * 
 * @copyright Copyright (c) 2021
 * 
 */
#include <iostream>
#include <queue>
using namespace std;

class RecentCounter {
public:
    RecentCounter() {

    }
    
    /*
    两组：
    add/remove/element
    offer/poll/peek
    联想C++ STL:
    push/pop/ front back
    */
    int ping(int t) {
        queue.push(t);
        while (!queue.empty() && (t - queue.front() > 3000)) {
            queue.pop();
        }
        return queue.size();
    }
private:
    queue<int> queue;
};

int main() {

    return 0;
}

File: 04_OJ/oj_leetcode_70/LC933.java
/*
@auther liuyang
@since 2021/12/2
参考：https://www.bilibili.com/read/cv8623235?from=articleDetail
OJ状态：TODO
*/
import java.util.LinkedList;
import java.util.Queue;

class LC933 {

    Queue<Integer> queue;

    public LC933() {
        queue = new LinkedList<Integer>();
    }

    /*
    两组：
    add/remove/element
    offer/poll/peek
    联想C++ STL:
    push/pop/ front back
    */
    public int ping(int i) {
        queue.add(i);
        while (!queue.isEmpty() && (t - queue.element() > 3000) {
            queue.remove();
        }
        return queue.size();
    }

}